app/gui/controller/double-representation/src/refactorings/collapse.rs

```diff
@@ -206,13 +206,13 @@ impl Extracted {
         let mut output = None;
         for connection in graph.info.connections() {
             let starts_inside = extracted_nodes_set.contains(&connection.source.node);
-            let ends_inside = extracted_nodes_set.contains(&connection.destination.node);
-            let identifier = graph.connection_variable(&connection)?;
+            let ends_inside = extracted_nodes_set.contains(&connection.target.node);
 
             leaves.remove(&connection.source.node);
             if !starts_inside && ends_inside {
-                inputs.push(identifier)
+                inputs.push(graph.connection_variable(&connection)?)
```

Comment:
Now, if we somehow cannot get `connection_variable`, the node won't be treated as a leaf anymore. Is that a conscious change?

Reply:
You are right, the logic change here was not intentional. I'll change it back. Though I'm not really sure which way is "more correct" here.